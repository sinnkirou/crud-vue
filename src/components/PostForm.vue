<template>
  <div id="PostForm">
    <div class="navbar">
      <h2 class="center">Post It</h2>
    </div>
    <div class="post_container" key="PostForm">
      <h1 class="post_heading">Create Post</h1>
      <EditablePost v-bind:posts="posts"/>
    </div>
  </div>
</template>

<script>
import EditablePost from "./EditablePost.vue";
export default {
  name: "PostForm",
  props: {
    posts: Array
  },
  components: {
    EditablePost
  }
};
</script>
