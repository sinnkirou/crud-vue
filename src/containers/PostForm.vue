<template>
  <div>
    <div class="navbar">
      <h2 class="center">Post It</h2>
    </div>
    <div class="post_container" key="PostForm">
      <h1 class="post_heading">Create Post</h1>
      <EditablePost :posts="posts"/>
    </div>
  </div>
</template>

<script>
export default {
  name: "PostForm",
  props: {
    posts: Array
  },
  methods: {},
  components: {
    EditablePost: () => import("../components/EditablePost")
  }
};
</script>
