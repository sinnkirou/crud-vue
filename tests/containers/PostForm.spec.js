import { shallowMount } from "@vue/test-utils";
import PostForm from "@/containers/PostForm.vue";

describe("PostForm.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PostForm);
  });

  it("should render", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
