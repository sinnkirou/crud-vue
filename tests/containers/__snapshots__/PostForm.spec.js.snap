// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostForm.vue has the expected html structure 1`] = `
<div>
  <div
    class="navbar"
  >
    <h2
      class="center"
    >
      Post It
    </h2>
  </div>
   
  <div
    class="post_container"
  >
    <h1
      class="post_heading"
    >
      Create Post
    </h1>
     
    <div
      class="post"
    >
      <form
        class="form"
      >
        <input
          placeholder="Enter Post Title"
          required="required"
          type="text"
        />
         
        <br />
         
        <br />
         
        <textarea
          cols="28"
          placeholder="Enter Post"
          required="required"
          rows="5"
        />
         
        <br />
         
        <br />
         
        <button
          class="mdl-button mdl-js-button mdl-button--raised"
          type="submit"
        >
          Create
        </button>
      </form>
    </div>
  </div>
</div>
`;
